lib/ansible/modules/database/postgresql/postgresql_idx.py

```diff
@@ -0,0 +1,411 @@
+            return False
+
+        cursor.execute(query)
+        # In any case, even the created index is not valid,
+        # the database schema has been changed:
+        changed = True
+    except psycopg2.InternalError as e:
+        if e.pgcode == '25006':
+            # Handle errors due to read-only transactions indicated by pgcode 25006
+            # ERROR:  cannot execute ALTER ROLE in a read-only transaction
+            changed = False
+            module.fail_json(msg=e.pgerror, exception=traceback.format_exc())
+            return changed
+        else:
+            raise psycopg2.InternalError(e)
+    return changed
+
+
+def index_drop(cursor, module, idxname, concurrent=True):
+    """Drop index"""
+    changed = False
+    if not index_exists(cursor, idxname):
+        return changed
+
+    mode = 'CONCURRENTLY'
+    if not concurrent:
+        mode = ''
+
+    query = 'DROP INDEX %s %s' % (mode, idxname)
+    try:
+        cursor.execute(query)
+        changed = True
+    except psycopg2.InternalError as e:
+        if e.pgcode == '25006':
+            # Handle errors due to read-only transactions indicated by pgcode 25006
+            # ERROR:  cannot execute ALTER ROLE in a read-only transaction
+            changed = False
+            module.fail_json(msg=e.pgerror, exception=traceback.format_exc())
+            return changed
+        else:
+            raise psycopg2.InternalError(e)
+    return changed
+
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    argument_spec = pgutils.postgres_common_argument_spec()
+    argument_spec.update(dict(
+        idxname=dict(required=True, aliases=['idxname']),
+        db=dict(default=''),
+        ssl_mode=dict(default='prefer', choices=[
+            'disable', 'allow', 'prefer', 'require', 'verify-ca', 'verify-full']),
+        ssl_rootcert=dict(default=None),
+        state=dict(default="present", choices=["absent", "present"]),
+        concurrent=dict(type=bool, default="yes"),
+        table=dict(default=None),
+        idxtype=dict(default=None),
+        columns=dict(default=None),
+        cond=dict(default=None)
+    ))
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    idxname = module.params["idxname"]
+    db = module.params["db"]
+    state = module.params["state"]
+    concurrent = module.params["concurrent"]
+    table = module.params["table"]
+    idxtype = module.params["idxtype"]
+    columns = module.params["columns"]
+    cond = module.params["cond"]
+    sslrootcert = module.params["ssl_rootcert"]
+
+    if state == 'present':
+        if table is None:
+            module.fail_json(msg="Table must be specified")
+        if columns is None:
+            module.fail_json(msg="At least one column must be specified")
+    else:
+        if table is not None:
+            module.fail_json(msg="Index is going to be removed, so "
```

Comment:
can you add the index name in the log ? It will permit to easily track the problem if we are using this module with a with_item ?

Reply:
It's a good idea, commited